### PR TITLE
AMPI: implement MPI_Win_{lock,unlock}_all

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

File: src/libs/ck-libs/ampi/ampiOneSided.C [hunk 1]
```diff
@@ -1021,6 +1021,52 @@ AMPI_API_IMPL(int, MPI_Win_unlock, int rank, MPI_Win win)
...
+ *   Returns int : MPI_SUCCESS or MPI_ERR_WIN
+ */
+AMPI_API_IMPL(int, MPI_Win_lock_all, int assert, MPI_Win win)
+{
+  AMPI_API("AMPI_Win_lock_all");
+  WinStruct *winStruct = getAmpiParent()->getWinStruct(win);
+  ampi *ptr = getAmpiInstance(winStruct->comm);
+  int size = ptr->getSize();
+
+  // process assertion here:
```
Comment (stwhite91) [inline src/libs/ck-libs/ampi/ampiOneSided.C]: Could maybe change these two lines to "// TODO: optimize for assertions"

File: src/libs/ck-libs/ampi/ampiOneSided.C [hunk 2]
```diff
@@ -1021,6 +1021,52 @@ AMPI_API_IMPL(int, MPI_Win_unlock, int rank, MPI_Win win)
...
+ */
+// The RMA call is completed both locally and remotely after unlock.
+AMPI_API_IMPL(int, MPI_Win_unlock_all, MPI_Win win)
+{
+  AMPI_API("AMPI_Win_unlock_all");
+  WinStruct *winStruct = getAmpiParent()->getWinStruct(win);
+  ampi *ptr = getAmpiInstance(winStruct->comm);
+  int size = ptr->getSize();
+
+  // process assertion here:
```
Comment (stwhite91) [inline src/libs/ck-libs/ampi/ampiOneSided.C]: There is no assert argument passed into Win_unlock_all

**Comment by stwhite91** [comment]
Could add error checking on the arguments passed into these routines, though I don't think we do uch if any error checking for the RMA routines currently

File: src/libs/ck-libs/ampi/ampiOneSided.C (hunk 1 above)
Comment (matthiasdiener) [reply src/libs/ck-libs/ampi/ampiOneSided.C]: fixed

File: src/libs/ck-libs/ampi/ampiOneSided.C (hunk 2 above)
Comment (matthiasdiener) [reply src/libs/ck-libs/ampi/ampiOneSided.C]: fixed.

**Comment by matthiasdiener** [comment]
> Could add error checking on the arguments passed into these routines, though I don't think we do uch if any error checking for the RMA routines currently

I added an (optional) check to getWinStruct.

**Pull request merged**